FreeIT/src/lesson_10/Application.java

```diff
@@ -71,41 +72,38 @@ private void printAllBooks() {
 		if (libr.getBooks() == null) {
 			System.out.println("В библиотеке нет книг. Нечего выводить!");
 		} else {
-			System.out.println(
-					"Выберите желаемую сортировку при выводе (введите число, соответствующее желаемой сортировке): \n1. По алфавиту по названию книги.\n2. По добавлению(сначала новые, потом более старые).\n3. По добавлению(сначала старые, потом новые).");
-			Scanner sc = new Scanner(System.in);
-			int choice = 0;
-			try {
-				choice = sc.nextInt();
-			} catch (InputMismatchException e) {
-				System.out.println("Вы должны ввести число! Попробуйте еще раз!");
-				printAllBooks();
-				return;
-			}
+			Integer choice = 0;
+			do {
+				System.out.println(
+						"Выберите желаемую сортировку при выводе (введите число, соответствующее желаемой сортировке): \n1. По алфавиту по названию книги.\n2. По добавлению(сначала новые, потом более старые).\n3. По добавлению(сначала старые, потом новые).");
+				choice = inputFromConcole();
+			} while (choice == null);
```

Comment:
choise will never be 'null' here. Remove current do-while cycle and left here only console input.